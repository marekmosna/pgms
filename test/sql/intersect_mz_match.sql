\set ECHO none
BEGIN;

\i test/pgtap-core.sql
\i sql/pgms.sql

SELECT plan(7);

SELECT is(
    intersect_mz(ref, query),
    expected,
    'intersect_mz should be' || to_char(expected, '0D99')
) FROM (VALUES
    (
        '{}'::spectrum, 
        '{}'::spectrum,
        0.0::float4
    ),
    (
        '{}'::spectrum, 
        '{{290}, {1.0}}'::spectrum,
        0.0::float4
    )
) v(ref, query, expected);

SELECT is(
    intersect_mz(ref, query),
    expected,
    'intersect_mz should be' || to_char(expected, '0D99')
) FROM (VALUES
    (
        '{{100, 200, 300,   500}, {1.0, 1.0, 1.0, 1.0}}'::spectrum, 
        '{{100, 200, 290, 499.9}, {1.0, 1.0, 1.0, 1.0}}'::spectrum,
        (3.00/4.00)::float4
    ),
    (
        '{{100, 200, 300,   500}, {1.0, 1.0, 1.0, 1.0}}'::spectrum, 
        '{{290}, {1.0}}'::spectrum,
        (0.00)::float4
    ),
    (
        '{{100, 200, 300,   500}, {1.0, 1.0, 1.0, 1.0}}'::spectrum, 
        '{{100, 200, 200.1, 499.9}, {1.0, 1.0, 1.0, 1.0}}'::spectrum,
        (3.00/4.00)::float4
    ),
    (
        '{{10 , 200, 300,   500}, {1.0, 1.0, 1.0, 1.0}}'::spectrum, 
        '{{100, 200, 200.1, 499.9}, {1.0, 1.0, 1.0, 1.0}}'::spectrum,
        (2.00/4.00)::float4
    ),
    (
        '{{100, 200, 200.1, 499.9}, {1.0, 1.0, 1.0, 1.0}}'::spectrum,
        '{{10 , 200, 300,   500}, {1.0, 1.0, 1.0, 1.0}}'::spectrum,
        (2.00/4.00)::float4
    )
) v(ref, query, expected);

SELECT * FROM finish();
ROLLBACK;
