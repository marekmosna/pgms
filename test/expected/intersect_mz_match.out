\set ECHO none
1..7
ok 1 - intersect_mz should be 0.00
ok 2 - intersect_mz should be 0.00
ok 3 - intersect_mz should be 0.75
ok 4 - intersect_mz should be 0.00
ok 5 - intersect_mz should be 0.75
ok 6 - intersect_mz should be 0.50
ok 7 - intersect_mz should be 0.50
